src/Blazored.Typeahead/Forms/BlazoredTypeaheadInput.razor.cs

```diff
@@ -14,22 +14,28 @@ public class BlazoredTypeaheadInputBase<TItem> : InputBase<TItem>, IDisposable
         [Inject] IJSRuntime JSRuntime { get; set; }
 
         [Parameter] public string Placeholder { get; set; }
-        [Parameter] public Func<string, Task<List<TItem>>> SearchMethod { get; set; }
+        [Parameter] public Func<string, Task<IEnumerable<TItem>>> SearchMethod { get; set; }
         [Parameter] public RenderFragment NotFoundTemplate { get; set; }
         [Parameter] public RenderFragment<TItem> ResultTemplate { get; set; }
         [Parameter] public RenderFragment<TItem> SelectedTemplate { get; set; }
         [Parameter] public RenderFragment FooterTemplate { get; set; }
         [Parameter] public int MinimumLength { get; set; } = 1;
         [Parameter] public int Debounce { get; set; } = 300;
+        [Parameter] public int MaximumSuggestions { get; set; } = 25;
 
         protected bool Searching { get; set; } = false;
-        protected bool EditMode { get; set; } = true;
-        protected List<TItem> SearchResults { get; set; } = new List<TItem>();
-
+        protected bool ShouldShowMenu { get; private set; } = false;
+        protected bool ShouldShowInput { get; private set; } = true;
```

Comment:
See comments on naming on the non-forms typeahead

Reply:
protected bool IsSearching { get; private set; } = false;
        protected bool IsShowingSuggestions { get; private set; } = false;
        protected bool IsShowingSearchbar { get; private set; } = true;
        protected bool IsShowingMask { get; private set; } = false;